inference-engine/src/vpu/common/src/ngraph/transformations/merge_gather_gather_elements.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "vpu/ngraph/transformations/merge_gather_gather_elements.hpp"
+
+#include <vpu/ngraph/operations/exp_gather_elements.hpp>
+#include <vpu/ngraph/utilities.hpp>
+
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/opsets/opset6.hpp>
+#include <ngraph/pattern/matcher.hpp>
+#include <ngraph/pattern/op/label.hpp>
+#include <ngraph/validation_util.hpp>
+#include <ngraph/log.hpp>
+
+#include <numeric>
+
+NGRAPH_RTTI_DEFINITION(vpu::MergeGatherGatherElements, "MergeGatherGatherElements", 0);
+
+namespace vpu {
+
+bool MergeGatherGatherElements::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    bool wasGraphChanged = false;
+
+    const auto gatherData = ngraph::pattern::any_input();
+    const auto gatherIndices = ngraph::pattern::any_input();
+    const auto gather = ngraph::pattern::wrap_type<ngraph::opset6::Gather>(
+        {gatherData, gatherIndices, ngraph::pattern::any_input()},
+        ngraph::pattern::consumers_count(1));
+
+    const auto squeezeAxes = ngraph::pattern::any_input(
+        ngraph::pattern::op::as_value_predicate(ngraph::pattern::has_class<ngraph::opset6::Constant>()));
+    const auto squeeze = ngraph::pattern::wrap_type<ngraph::opset6::Squeeze>({gather, squeezeAxes}, ngraph::pattern::consumers_count(1));
+
+    const auto transposePerm = ngraph::pattern::any_input(
+        ngraph::pattern::op::as_value_predicate(ngraph::pattern::has_class<ngraph::opset6::Constant>()));
+    const auto transpose = ngraph::pattern::wrap_type<ngraph::opset6::Transpose>({squeeze, transposePerm});
+
+    const auto m = std::make_shared<ngraph::pattern::Matcher>(transpose, "GatherSqueezeTransposeMatcher");
+    for (const auto& node : f->get_ordered_ops()) {
+        if (m->match(node)) {
```

Comment:
Nitpick: save level of indentation

```suggestion
        if (!m->match(node)) {
            continue;
        }

        // else branch
```

Reply:
Thanks! Done.